### PR TITLE
feat: add toast messages component

### PR DESCRIPTION
**Pull request by muubar**
## Description [[STORY LINK]](https://app.asana.com/0/1199393268587978/1200482301351717)

- Add and export a new component called "toast", which is a wrapper around [React-hot-toast's](https://react-hot-toast.com/) toast function and Toaster component. please check their docs for details. to show a toast, import the `toast` function, and call `toast.error` or `toast.success`, you can also pass more options and/or override the default configuration we're using.

- Show an error toast if the user is clicking on an e-commerce item's purchase button and the item's currency doesn't match the user account's currency

- Use a slightly better solid X icon, and make the alert's close button a bit smaller

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Tested changes locally.
- [ ] Updated documentation.

## Screenshots

alert icons and button
toast screenshot

### CODE REVIEW

**Comment by Marwan-Pelcro** [comment]
🔥 🔥 🔥 🔥 🔥 🔥

File: src/Main.js [hunk 1]
```diff
@@ -60,6 +61,8 @@ export const Main = () => {
       <CartModal />
       <OrderCreateModal />
       <OrderConfirmModal />
+
+      <Toaster />
```
Comment (ProgaPanda) [inline src/Main.js:65]: Should this be optional? Or part of the modal controller render function?

File: src/Components/common/PelcroNativeButtons/index.js [hunk 2]
```diff
@@ -242,20 +243,31 @@ export const init = () => {
         pelcroPurchaseButtonsByClass[i].addEventListener(
           "click",
           (e) => {
-            const skuIdToPurcahse = Number(e.target.dataset.skuId);
+            const itemSkuId = Number(e.target.dataset.skuId);
+            const quickPurchaseItem =
+              window.Pelcro.ecommerce.products.getBySkuId(
+                Number(itemSkuId)
```
Comment (ProgaPanda) [inline src/Components/common/PelcroNativeButtons/index.js:249]: You coerced it into a number already ln:246

**Review by ProgaPanda — COMMENTED** [review]
Nice 🔥

File: src/Main.js (hunk 1 above)
Comment (muubar) [reply src/Main.js:65]: it's already part of the `Controller` comp render, I'm using the same approach we always use, searching in children for a `viewId` of `toaster` and rendering it if it's there

Comment (ProgaPanda) [reply src/Main.js:65]: Yeah, but users will need to add `<Toaster />` for it to work. I'm talking about whether we should add it to be rendered by ModalController anyway. It's not a button or anything

Comment (muubar) [reply src/Main.js:65]: I completely agree with you, but I also wanted to respect the fact that the `Controller` comp is only a controller and doesn't force anything to be rendered, if the controller renders something as part of it's implementation then it's no longer a controller, so I preferred to respect the design over the inconvenience of adding the component

**Pull request merged**